coding-the-legislation/37_inferences.md

```diff
@@ -0,0 +1,26 @@
+# Inferences
+
+Here are the places in which inferences take place in OpenFisca:
+
+- In parameters values, through the `fuzzy` attribute, which artificially extends the validity period of a value.
+- In period management, through `calculate_output`, which can automatically sum or divide values over periods to match the requested period.
+- In period management, through `set_input`, which can automatically sum or divide input values over periods to match the computable period.
+- In some formulas, through `base_function`, which can yield values that the original requested formula could not compute on its own.
+- In some formulas, through `max_nb_cycles`, which can block the computation toward the past and thus not allow some values to be computed.
+
+These inferences are not considered good practice, as they tend to make computations less consistent and predictable. You should tend to avoid relying on them as much as possible.
+
+
+## Known issues
+
+### With `fuzzy`
+
+This parameter introduces a major production risk: if the end date of a parameter used in a computation is reached, then the program will crash.
+
+The advised workaround is to _always_ add the `fuzzy` parameter to the last known value.
+
+### With `base_function`
+
+Default values in input variables are the source of the issue: if a formula needs values that are undefined (e.g. because they are in previous months from the calculation), it won't crash nor log anything because the input variables return their default values. This inference behaviour seems to be fine for macrosimulation, but production would expect a time-based inference, where the value is copied from its closest defined value rather than a global default one.
```

Comment:
I won't use the word macrosimulation because what I do is indeed microsimulation. 
Macrosimulation sounds like macroeconomics and we are actually doing microsimulation because we simulate micro-variables that depend on the micro characteristics of individuals or micro entities.
I would say microsimulation with survey data, or population data, or array data or non test-case simulation definitively not macrosimulation